node/impl/full/eth_trace.go

```diff
@@ -0,0 +1,339 @@
+	if et.Msg.ReadOnly {
+		trace.SetCallType("staticcall")
+	}
+
+	// there are several edge cases thar require special handling when displaying the traces
+	if parent != nil {
+		// Handle Native actor creation
+		//
+		// Actor A calls to the init actor on method 2 and The init actor creates the target actor B then calls it on method 1
+		if parent.Action.FilecoinTo == builtin.InitActorAddr &&
+			parent.Action.FilecoinMethod == builtin.MethodsInit.Exec &&
+			et.Msg.Method == builtin.MethodConstructor {
+			log.Debugf("Native actor creation! method:%d, code:%s, height:%d", et.Msg.Method, et.Msg.CodeCid.String(), height)
+			parent.SetCallType("create")
+			parent.Action.To = et.Msg.To.String()
+			parent.Action.Input = []byte{0x0, 0x0, 0x0, 0xFE}
+			parent.Result.Output = nil
+
+			// there should never be any subcalls when creating a native actor
+			return nil
+		}
+
+		// Handle EVM contract creation
+		//
+		// To detect EVM contract creation we need to check for the following sequence of events:
+		//
+		// 1) EVM contract A calls the EAM (Ethereum Address Manager) on method 2 (create) or 3 (create2).
+		// 2) The EAM calls the init actor on method 3 (Exec4).
+		// 3) The init actor creates the target actor B then calls it on method 1.
+		if parent.Parent != nil {
+			calledCreateOnEAM := parent.Parent.Action.FilecoinTo == builtin.EthereumAddressManagerActorAddr &&
+				(parent.Parent.Action.FilecoinMethod == builtin.MethodsEAM.Create || parent.Parent.Action.FilecoinMethod == builtin.MethodsEAM.Create2)
+			eamCalledInitOnExec4 := parent.Action.FilecoinTo == builtin.InitActorAddr &&
+				parent.Action.FilecoinMethod == builtin.MethodsInit.Exec4
+			initCreatedActor := trace.Action.FilecoinMethod == builtin.MethodConstructor
+
+			if calledCreateOnEAM && eamCalledInitOnExec4 && initCreatedActor {
+				log.Debugf("EVM contract creation method:%d, code:%s, height:%d", et.Msg.Method, et.Msg.CodeCid.String(), height)
+
+				if parent.Parent.Action.FilecoinMethod == builtin.MethodsEAM.Create {
+					parent.Parent.SetCallType("create")
+				} else {
+					parent.Parent.SetCallType("create2")
+				}
+
+				// update the parent.parent to make this
+				parent.Parent.Action.To = trace.Action.To
+				parent.Parent.Subtraces = 0
+
+				// delete the parent (the EAM) and skip the current trace (init)
+				*traces = (*traces)[:len(*traces)-1]
+
+				return nil
+			}
+		}
+
+		// Handle EVM call special casing
+		//
+		// Any outbound call from an EVM actor on methods 1-1023 are side-effects from EVM instructions
+		// and should be dropped from the trace.
+		if builtinactors.IsEvmActor(parent.Action.FilecoinCodeCid) &&
+			et.Msg.Method > 0 &&
+			et.Msg.Method <= 1023 {
+			log.Debugf("found outbound call from an EVM actor on method 1-1023 method:%d, code:%s, height:%d", et.Msg.Method, parent.Action.FilecoinCodeCid.String(), height)
+			// TODO: if I handle this case and drop this call from the trace then I am not able to detect delegate calls
```

Comment:
At a minimum, drop this call.

Reply:
It could be GetBytecode, GetBytecodeHash, etc. I.e., something internal.

Reply:
Some field in the parent for "last getbytecode"?

Reply:
I.e., need to drop all these calls _but_ we also need to be able to lookup the "last" GetBytecode when we do the delegate call.

Reply:
I.e., we need to record `CalledGetBytecodeOn`

Reply:
And remember to skip all the children of this call as well. I.e., we'll _always_ early-return after handling this call.